### PR TITLE
chore: add dictionary testing

### PR DESCRIPTION
**Pull request by pgautier404**

### CODE REVIEW

File: momento/dictionary_test.go [hunk 1]
```diff
@@ -0,0 +1,676 @@
...
+			).Error().To(BeNil())
+			getFieldResp, err := sharedContext.Client.DictionaryGetField(sharedContext.Ctx, &DictionaryGetFieldRequest{
+				CacheName:      sharedContext.CacheName,
+				DictionaryName: sharedContext.CollectionName,
+				Field:          field,
+			})
+			Expect(err).To(BeNil())
+			Expect(getFieldResp).To(BeAssignableToTypeOf(&DictionaryGetFieldHit{}))
+			switch result := getFieldResp.(type) {
+			case *DictionaryGetFieldHit:
```
Comment (eaddingtonwhite) [inline momento/dictionary_test.go:119]: should we add default on this switch w/ an explicit fail if hits?

File: momento/dictionary_test.go [hunk 2]
```diff
@@ -0,0 +1,676 @@
...
+
+	It("returns an error for set field when field is empty", func() {
+		Expect(
+			sharedContext.Client.DictionarySetField(sharedContext.Ctx, &DictionarySetFieldRequest{
+				CacheName:      sharedContext.CacheName,
+				DictionaryName: sharedContext.CollectionName,
+				Field:          String(""),
+				Value:          String("myValue"),
+			}),
+		).Error().To(HaveMomentoErrorCode(InvalidArgumentError))
```
Comment (eaddingtonwhite) [inline momento/dictionary_test.go:140]: maybe add case for nil also? I had made this [ticket](https://github.com/momentohq/client-sdk-go/issues/178) for nil `Field` or `Value` also so can wait till we do that if easier come back in one pass.

File: momento/dictionary_test.go [hunk 3]
```diff
@@ -0,0 +1,676 @@
...
+
+	It("returns an error if an item field is empty", func() {
+		Expect(
+			sharedContext.Client.DictionarySetFields(sharedContext.Ctx, &DictionarySetFieldsRequest{
+				CacheName:      sharedContext.CacheName,
+				DictionaryName: sharedContext.CollectionName,
+				Items:          map[string]Value{"myField": String("myValue"), "": String("myOtherValue")},
+			}),
+		).Error().To(HaveMomentoErrorCode(InvalidArgumentError))
+	})
```
Comment (eaddingtonwhite) [inline momento/dictionary_test.go:194]: same here for nil check. Maybe just add TODO comment in test for now?

**Review by eaddingtonwhite — APPROVED** [review]
Awesome job on these @pgautier404 !! Loving the new test lib were using looks great.

File: momento/dictionary_test.go (hunk 1 above)
Comment (pgautier404) [reply momento/dictionary_test.go:119]: I've already asserted that it is assignable to that type above. The only purpose for the switch is to get the cast.

File: momento/dictionary_test.go (hunk 2 above)
Comment (pgautier404) [reply momento/dictionary_test.go:140]: Good call, I'll add that in a followup.

File: momento/dictionary_test.go (hunk 3 above)
Comment (pgautier404) [reply momento/dictionary_test.go:194]: Same, will follow up on this in my list testing PR.

**Pull request merged**